inference-engine/src/vpu/graph_transformer/src/stages/gather_nd.cpp

```diff
@@ -0,0 +1,100 @@
+//
+
+#include <vpu/frontend/frontend.hpp>
+
+#include <memory>
+#include <string>
+
+namespace vpu {
+
+namespace {
+
+class GatherNDStage final : public StageNode {
+public:
+    using StageNode::StageNode;
+
+protected:
+    StagePtr cloneImpl() const override { return std::make_shared<GatherNDStage>(*this); }
+
+    void propagateDataOrderImpl(StageDataInfo<DimsOrder> &orderInfo) override {
+        const auto input1 = inputEdge(0)->input();
+        const auto input2 = inputEdge(1)->input();
+        const auto output = outputEdge(0)->output();
+
+        orderInfo.setInput(inputEdge(0), DimsOrder::fromNumDims(input1->desc().numDims()));
+        orderInfo.setInput(inputEdge(1), DimsOrder::fromNumDims(input2->desc().numDims()));
+        orderInfo.setOutput(outputEdge(0), DimsOrder::fromNumDims(output->desc().numDims()));
+    }
+
+    void getDataStridesRequirementsImpl(StageDataInfo<StridesRequirement> &stridesInfo) override {
+        for (const auto &inEdge : inputEdges()) {
+            stridesInfo.setInput(inEdge, StridesRequirement::compact());
+        }
+        stridesInfo.setOutput(outputEdge(0), StridesRequirement::compact());
+    }
+
+    void finalizeDataLayoutImpl() override {}
+
+    void getBatchSupportInfoImpl(StageDataInfo<BatchSupport> &batchInfo) override {}
+
+    StageSHAVEsRequirements getSHAVEsRequirementsImpl() const override {
+        return StageSHAVEsRequirements::NotNeeded;
+    }
+
+    void initialCheckImpl() const override {
+        IE_ASSERT(numInputs() == 2);
+        IE_ASSERT(numOutputs() == 1);
+        assertInputsOutputsTypes(
+            this,
+            {{DataType::U8, DataType::FP16, DataType::S32}, {DataType::S32}},
+            {{DataType::U8, DataType::FP16, DataType::S32}});
+    }
+
+    void serializeParamsImpl(BlobSerializer &serializer) const override {
+        const auto batch_dims = attrs().get<int32_t>("batch_dims");
+        serializer.append(batch_dims);
+    }
+
+    void serializeDataImpl(BlobSerializer &serializer) const override {
+        auto input0 = inputEdge(0)->input();
+        auto input1 = inputEdge(1)->input();
+        auto output = outputEdge(0)->output();
+
+        input0->serializeBuffer(serializer);
+        output->serializeBuffer(serializer);
+        input1->serializeBuffer(serializer);
+    }
+};
+
+} // namespace
+
+void FrontEnd::parseGatherND(
+    const Model &model,
+    const ie::CNNLayerPtr &layer,
+    const DataVector &inputs,
+    const DataVector &outputs) const {
+    VPU_THROW_UNLESS(
+        inputs.size() == 2,
+        "GatherND parsing failed, expected number of inputs: 2, but {} provided.",
+        inputs.size());
+    VPU_THROW_UNLESS(
+        outputs.size() == 1,
+        "GatherND parsing failed, expected number of outputs: 1, but {} provided.",
+        outputs.size());
+    VPU_THROW_UNLESS(layer, "CNNLayer pointer is null.");
+
+    const auto batch_dims = layer->GetParamAsInt("batch_dims");
```

Comment:
Is it true that `batch_dims` param always exists?

Reply:
Constructor of GatherND operation has that value = 0 by default. Added it here also.